### PR TITLE
Auth failure changes

### PR DESCRIPTION
**Pull request by VinayKumar1591**
[5G SA: MVC][component: agw] [type: enhancement] 5G Skipping Identity Procedure

## Summary

1. Authentication MAC failure handling changes

## Test Plan

1. Tested on NGAP TESTER
2. Authentication Reject has been sent after MAC failure if initial Identity type is SUCI

### CODE REVIEW

File: lte/gateway/c/oai/tasks/amf/amf_as.cpp [hunk 1]
```diff
@@ -1248,6 +1275,106 @@ static int amf_as_security_req(
...
+ **          Others:   None                                                **
+ **                                                                        **
+ ***************************************************************************/
+static int amf_as_security_rej(
+    const amf_as_security_t* msg, m5g_dl_info_transfer_req_t* as_msg) {
+  OAILOG_FUNC_IN(LOG_NAS_AMF);
+  int size = 0;
+  amf_nas_message_t nas_msg;
+
+  memset(&nas_msg, 0, sizeof(amf_nas_message_t));
```
Comment (ulaskozat) [inline lte/gateway/c/oai/tasks/amf/amf_as.cpp]: `amf_nas_message_t nas_msg = {0}`

File: lte/gateway/c/oai/tasks/amf/amf_authentication.cpp [hunk 2]
```diff
@@ -647,6 +643,39 @@ int amf_proc_authentication_failure(
...
+         */
+        amf_sap_t amf_sap;
+        amf_sap.primitive                    = AMFAS_SECURITY_REJ;
+        amf_sap.u.amf_as.u.security.ue_id    = ue_id;
+        amf_sap.u.amf_as.u.security.msg_type = AMF_AS_MSG_TYPE_AUTH;
+        rc                                   = amf_sap_send(&amf_sap);
+      }
+    } break;
+    case AMF_CAUSE_SYNCH_FAILURE: {
+      // hanlde SYNCH Failure scenario
```
Comment (ulaskozat) [inline lte/gateway/c/oai/tasks/amf/amf_authentication.cpp]: typo `hanlde`

**Review by ulaskozat — APPROVED** [review]
Left some nit comments.

File: lte/gateway/c/oai/tasks/amf/amf_as.cpp (hunk 1 above)
Comment (VinayKumar1591) [reply lte/gateway/c/oai/tasks/amf/amf_as.cpp]: Done

File: lte/gateway/c/oai/tasks/amf/amf_authentication.cpp (hunk 2 above)
Comment (VinayKumar1591) [reply lte/gateway/c/oai/tasks/amf/amf_authentication.cpp]: Done

**Pull request merged**